app/models/search/tag_search_form.rb

```diff
@@ -36,6 +38,22 @@ def search_results
     @searcher.search_results
   end
 
+  def set_canonical
```

Comment:
Splitting the handling of `wrangling_status` into 2 methods makes it harder to follow what option is set in which case. Can you combine them into `set_wrangling_status` and use a case statement for `wrangling_status`?